lib/services/role.go

```diff
@@ -2210,3 +2214,34 @@ func MarshalRole(role types.Role, opts ...MarshalOption) ([]byte, error) {
 		return nil, trace.BadParameter("unrecognized role version %T", role)
 	}
 }
+
+// DowngradeToV3 converts a V4 role to V3 in-place so that it will be compatible with older instances.
+// DELETE IN 8.0.0
+func DowngradeRoleToV3(r *types.RoleV3) error {
```

Comment:
Nit: Not sure if its *necessary*, but I'd feel better if the downgrade process returned a new value that was a shallow copy in the event that the downgrade needed to occur.  To my knowledge, outbound resources aren't usually written to after their defaults are set.

Reply:
thanks for the review. I agree, I've updated this in the latest commit